tg_bot/modules/locks.py

```diff
@@ -199,6 +255,16 @@ def del_lockables(bot: Bot, update: Update):
                         chat.kick_member(new_mem.id)
                         message.reply_text("Only admins are allowed to add bots to this chat! Get outta here.")
             else:
+                #allow whitelisted URLs
+                if lockable == 'url':
+                    entities = set(url for url in message.parse_entities(MessageEntity.URL).values())
+                    #MessageEntity.TEXT_LINK could be added in the filter above, but would return the text, not the url,
+                    #so add all entities that have a 'url' field separately
+                    entities = entities | set(entity.url for entity in message.entities if entity.url)
```

Comment:
do not use `message.entities`; its not recommended in the docs. use parse_entities() instead

Reply:
The reason here is explained in the commented text. The `parse_entities()` method returns the text of the message that contains the link. It's fine for URLs sent directly in the text, but if it's the text link `[nice word](www.bad_link.com)`, it would return `"nice word"`.

All entities of the TEXT_LINK type (and **only** the entities of that type, AFAIK) have a `.url` attribute, and it contains the actual URL. So by adding all those URLs to the set, I'm sure I'm including the text link URLs as well.